Essentials/src/com/earth2me/essentials/Worth.java

```diff
@@ -62,6 +62,86 @@ public BigDecimal getPrice(IEssentials ess, ItemStack itemStack) {
         return result;
     }
 
+    /**
+     * Get the value of an item stack from the config for buying.
+     *
+     * @param ess       The Essentials instance.
+     * @param itemStack The item stack to look up in the config.
+     * @return The price from the config.
+     */
+    public BigDecimal getBuyPrice(IEssentials ess, ItemStack itemStack) {
```

Comment:
I feel like this whole function (as well as getSellPrice) can be folded into getPrice() where getPrice() has an additional argument for which 'keyspace' it is looking up the price in i.e. the 'buy' keyspace vs the 'worth' keyspace.

Reply:
Totally agree.  I just refactored this.